### DISCUSSION TITLE
Grafana UI: select-widget does not allow editing the selected value

### DISCUSSION THREAD

**Discussion by babs** [grafana/grafana#41093] [Plugins]
**Steps to reproduce**:
Create or edit a dashboard using InfluxQL editor on an influxdb datasource.
Select any field or value in FROM or WHERE clause, reselect the same field or value
Ex:
=> FROM `cpu` WHERE `host` = `xxxxx`
Reselect `xxxxx` (or `host` or `cpu`)

**Expected Result**
Field enters in edit mode (free to edit with the selected value pre-filled / modify field value)

**Actual Result**:
Field is set like the first time, no in-place edition possible

**Note**:
It was very convenient when you wanted to derive a regexp from a long hostname without having to type it in:
  example from `a-very-long-service-cluster01-node01` to `/a-very-long-service-cluster01-node\d+/` .

**Environment**:
- Grafana version: 8.1.4
- Data source type & version: InfluxQL // influxdb 1.x
- User OS & Browser: Xubuntu using chromium
- Others: is working on 7.x

**Comment by gabor** [comment]
@babs thanks for reporting the issue and describing your use-case... this functionality is very hard to discover, so i do not think we will restore it as it was, but we will try to find a way to support the use-case. one idea is to not clear the text-field when you click on it, so you could still modify it.

**Comment by babs** [comment]
Exactly, you got the idea.

**Comment by gabor** [comment]
unfortunately there is no easy way to do this.  the widget we use here ( https://developers.grafana.com/ui/latest/index.html?path=/story/forms-select--custom-value-creation ) does not support this.

i'll describe the workflow we need to support, using the widget from the above-mentioned link:
- i click on the select, and i choose an option, let’s say i choose “Beau Bevel”
- now i click into the widget again, the current value of the widget is “Beau Bevel”, and the cursor is at the end of the text
- i want to add the character “2” to the end of the currently chosen text, modifying it to “Beau Bevel2"

this is  not possible with the select that is on the linked page. i have to start typing, which clears the field, and type the whole “Beau Bevel2” manually. this is very annoying when one wants to do a small modification to the text and the text is very long.

there is a github-issue for the react-select widget with some possible workaround, but they don't seem simple: https://github.com/JedWatson/react-select/issues/1558

@grafana/user-essentials & @grafana/plugins-platform-frontend what do you think?

**Comment by gabor** [comment]
@grafana/user-essentials passing this to you for now 👍

**Comment by ashharrison90** [comment]
hmmmm, not sure if this is intended behaviour or not. although editing makes sense in your specific scenario, i can think of scenarios where this would be more annoying. we'll discuss as a squad tomorrow.

either way, specifics of the `Select` is in the `@grafana/ui` domain which is @grafana/plugins-platform responsibility i believe.

**Comment by gabor** [comment]
> hmmmm, not sure if this is intended behaviour or not. although editing makes sense in your specific scenario, i can think of scenarios where this would be more annoying. we'll discuss as a squad tomorrow.

yes, i agree that it is perhaps not useful for every situation.. i just would like to know if we consider adding it perhaps optionally (one more boolean-prop solves everything 😁 ), or we decide to not do it.

> either way, specifics of the `Select` is in the `@grafana/ui` domain which is @grafana/plugins-platform responsibility i believe.

i see. we should probably update the CODEOWNERS file to be more granular, right now the best match in that file is this pattern: https://github.com/grafana/grafana/blob/main/.github/CODEOWNERS#L74, and that one specifies 3 squads.

**Comment by hugohaggmark** [comment]
> i see. we should probably update the CODEOWNERS file to be more granular, right now the best match in that file is this pattern: https://github.com/grafana/grafana/blob/main/.github/CODEOWNERS#L74, and that one specifies 3 squads.

We're updating as we move forward, there are 3 squads set as owners for `/packages` because it used to be 1 squad which has split into 3. 😅

**Comment by natellium** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
